### ISSUE TITLE
Wrong PIN for RGB LED on board lolin_s3_mini

### ISSUE THREAD

**Issue by mp-se** [espressif/arduino-esp32#9037]
### Board

lolin_s3_mini

### Device Description

Its not possible to control the RGB led on this board with the neopixelWrite() function since the PIN number is wrong in the pins_arduino.h

The pin definition for this board is
``#define RGB_BUILTIN LED_BUILTIN``

In neopixelWrite the following operation is done
```
#ifdef RGB_BUILTIN
  if(pin == RGB_BUILTIN){
    _pin = RGB_BUILTIN-SOC_GPIO_PIN_COUNT;
  }
#endif
```

So the for this to work the definition for this board needs to be 
``#define RGB_BUILTIN LED_BUILTIN+SOC_GPIO_PIN_COUNT``

Not sure if the macro or the LED_BUILTIN constant should be changed so I put it as a bug report instead of PR.

If this is changed its possible to change the rgb led color on this board. 

### Hardware Configuration

Just plain board not connections

### Version

v2.0.14

### IDE Name

PlatformIO

### Operating System

Windows 11

### Flash frequency

40

### PSRAM enabled

yes

### Upload speed

115200

### Description

It should be possible to control the RGB led with neopixel on this board.

### Sketch

```cpp
#include <Arduino.h>

void setup() {
  neopixelWrite(RGB_BUILTIN, 255, 0, 0);
  delay(1000);
  neopixelWrite(RGB_BUILTIN, 0, 255, 0);
  delay(1000);
  neopixelWrite(RGB_BUILTIN, 0, 0, 255);
}

void loop() {
}
```

### Debug Message

```plain
E (240) rmt: rmt_set_gpio(526): RMT GPIO ERROR
E (240) rmt: rmt_config(686): set gpio for RMT driver failed
[   331][D][esp32-hal-rmt.c:615] rmtInit():  -- TX RMT - CH 0 - 1 RAM Blocks - pin 254
[   337][E][esp32-hal-rmt.c:631] rmtInit(): RMT failed to initilize.
[   343][E][esp32-hal-rgb-led.c:18] neopixelWrite(): RGB LED driver initialization failed!
```

### I have checked existing issues, online documentation and the Troubleshooting Guide

- [X] I confirm I have checked existing issues, online documentation and Troubleshooting guide.

**Comment by Eric-IOT** [comment]
hello, 
**SIMILIAR BEHAVIOUR**
running IDE on windows 10 :
Version: 2.3.0
Date: 2024-02-07T14:04:32.754Z
CLI Version: 0.35.2
and
Adafruit NeoPixel version 1.12.0

i get similar warnings when loading code, but the code is running

In the pin definition for LOLIN S3mini,  i made the proposed changes of @mp-se and the warning have disapeared.

**Comment by mp-se** [comment]
The change has already been implemented for the esp32s3 board here https://github.com/espressif/arduino-esp32/blob/master/variants/esp32s3/pins_arduino.h. So it has partly been implemented for some of the s3 boards but this should apply for all boards based on the s3 variant.

**Comment by einglis** [comment]
I believe the colour order is different too?  The example sketch `BlinkRGB` should go:
* White, Red, Green, Blue, ...
but actually goes:
* White, **Green**, **Red**, Blue, ...

**Comment by SuGlider** [comment]
This has been fixed for the Arduino Core 3.0.0
https://github.com/espressif/arduino-esp32/blob/master/variants/lolin_s3_mini/pins_arduino.h

**Comment by SuGlider** [comment]
We are close to release Arduino 2.0.15. 
Therefore, we may submit a PR to `release/v2.x` branch in order to apply changes for all the board.

**Comment by SuGlider** [comment]
> We are close to release Arduino 2.0.15. Therefore, we may submit a PR to `release/v2.x` branch in order to apply changes for all the board.

PR #9381 will add/backport all changes to 2.0.15

**Comment by SuGlider** [comment]
Fixed by #9381

**Comment by profharris** [comment]
Trying to fix the NeoPixel GRB color output instead of the expected RGB, 
I tried adding the line recommended above:
     "#define RGB_BUILTIN LED_BUILTIN+SOC_GPIO_PIN_COUNT"

That line locked up my LOLIN S3 Mini Pro.  I had to do the Hold down BOOT (Left Button),
Press and release the [Reset] Button, release the [BOOT] button... to put my S3 Mini Pro
into “Bootloader Mode”. Then remove the line, and re-upload my sketch.  

Perhaps it wasn't locked up, maybe the line directed RGB_BUILTIN to somewhere 
other than Pin 8, so the output was being sent who knows where. -profharris
